test/integration/basic_monitor_crossconnect_test.go

```diff
@@ -33,11 +33,11 @@ func TestSingleCrossConnect(t *testing.T) {
 	kubetest.DeployNSC(k8s, nodes[0].Node, "nsc-1", defaultTimeout)
 
 	// monitor client for node0
-	eventCh0, closeFunc0 := kubetest.XconProxyMonitor(k8s, nodes[0], "0")
+	eventCh0, closeFunc0 := kubetest.CrossConnectClientAt(k8s, nodes[0].Nsmd)
```

Comment:
Should these changes be done i a separate PR?

Reply:
I introduced `XconProxyMonitor` in #1357 and that changes is just reverting and replacing it with more general solution - `spire-proxy`. I don't think moving `spire-proxy` to separate PR makes sense